### PR TITLE
feat(nodejs): feature parity [1/N] - remote table

### PR DESCRIPTION
**Pull request by universalmind303**
closes https://github.com/lancedb/lancedb/issues/1362

### CODE REVIEW

File: nodejs/lancedb/remote/connection.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  }
+
+  async createTable(
+    tableName: string,
+    data: Data,
+    options?: Partial<CreateTableOptions> | undefined,
+  ): Promise<Table> {
+    if (options?.mode) {
+      console.warn(`mode is not supported in remote connection, ignoring.`);
+    }
```
Comment (wjones127) [inline nodejs/lancedb/remote/connection.ts:115]: Could you describe what mode behaviour it is following here? It it going to error if the table already exists? Or overwrite the table?

File: nodejs/lancedb/remote/table.ts [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  display(): string {
+    return `RemoteTable(${this.#dbName}; ${this.#name})`;
+  }
+
+  async schema(): Promise<import("apache-arrow").Schema> {
+    const resp = await this.#client.post(`${this.#tablePrefix}/describe/`);
+    // TODO: parse this into a valid arrow schema
+    return resp.schema;
```
Comment (wjones127) [inline nodejs/lancedb/remote/table.ts:63]: Is this handled in one of the follow-up PRs?

**Review by wjones127 — COMMENTED** [review]
Good work! I have a couple minor comments.

File: nodejs/lancedb/remote/table.ts (hunk 2 above)
Comment (universalmind303) [reply nodejs/lancedb/remote/table.ts:63]: I don't have this handled yet.

The old `vectorDB` sdk was returning this AS IS and silently deviating from the type hints. Currently this mimics the same functionality, but I wanted to call out that this is not the correct behavior.

File: nodejs/lancedb/remote/connection.ts (hunk 1 above)
Comment (universalmind303) [reply nodejs/lancedb/remote/connection.ts:115]: We can definitely make this a bit more informative. _FWIW , I was copying the python message_

I'd assume cloud behavior is to error on exists?

File: nodejs/lancedb/remote/table.ts (hunk 2 above)
Comment (wjones127) [reply nodejs/lancedb/remote/table.ts:63]: Okay. This is fine for now.

**Review by wjones127 — APPROVED** [review]

**Pull request merged**